src/test/mempool_tests.cpp

```diff
@@ -761,7 +762,7 @@ BOOST_AUTO_TEST_CASE(MempoolAncestryTests)
     tb = make_tx(/* output_values */ {5 * COIN, 3 * COIN}, /* inputs */  {ta});
     tc = make_tx(/* output_values */ {2 * COIN}, /* inputs */ {tb}, /* input_indices */ {1});
     td = make_tx(/* output_values */ {6 * COIN}, /* inputs */ {tb, tc}, /* input_indices */ {0, 0});
-    pool.clear();
+    pool.clearTxs();
```

Comment:
What is the difference between clearing and just instantiating a new mempool?

Reply:
No difference. Though, a new mempool needs the old `cs` released and the fresh `cs  taken, so would be a bit more code

Reply:
> needs the old `cs` released

Why?

Reply:
Ok, it is not needed if the scope of `pool` is kept the same. Though, it is still a larger diff to instantiate a new mempool `pool2`, lock it and replace `pool` with `pool2`. Happy to do whatever reviewers want, so let me know if I should keep this or change it.

Reply:
I was just asking. It doesn't need to be a big change, something like:
```
{
TxMemPoolClearable pool;
// ...
}

{
TxMemPoolClearable pool;
// ...
}
```

Reply:
thx, fixed